daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/grpcLedgerClient/AdminLedgerClient.scala

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Temporary stand-in for the real admin api clients defined in canton. Needed only for upgrades testing
+// We should intend to replace this as soon as possible
+package com.daml.lf.engine.script.v2.ledgerinteraction
+package grpcLedgerClient
+
+import com.daml.ledger.api.auth.client.LedgerCallCredentials.authenticatingStub
+import com.daml.ledger.client.configuration.LedgerClientChannelConfiguration
+import com.daml.ledger.client.GrpcChannel
+import com.daml.lf.engine.script.admin_package_service
+import io.grpc.Channel
+import io.grpc.netty.NettyChannelBuilder
+import io.grpc.stub.AbstractStub
+import java.io.Closeable
+import scala.concurrent.{ExecutionContext, Future}
+
+final class AdminLedgerClient private (
+    val channel: Channel,
+    token: Option[String],
+)(implicit ec: ExecutionContext)
+    extends Closeable {
+
+  private val packageServiceStub =
+    AdminLedgerClient.stub(admin_package_service.PackageServiceGrpc.stub(channel), token)
+
+  def vetDarByHash(darHash: String): Future[Unit] =
+    packageServiceStub.vetDar(admin_package_service.VetDarRequest(darHash, true)).map(_ => ())
+
+  def unvetDarByHash(darHash: String): Future[Unit] =
+    packageServiceStub.unvetDar(admin_package_service.UnvetDarRequest(darHash)).map(_ => ())
+
+  // Gets all (first 1000) dar names and hashes
+  def listDars(): Future[Seq[(String, String)]] =
+    packageServiceStub
+      .listDars(admin_package_service.ListDarsRequest(1000))
+      .map(res => res.dars.map(darDesc => (darDesc.name, darDesc.hash)))
```

Comment:
Should we not failed or emit a warning if more there are 1000 dars ?

Reply:
Fair point